### ISSUE TITLE
Page context is sometimes wrong

### ISSUE THREAD

**Issue by falkoschindler** [zauberzeug/nicegui#70]
In the following example we create a scene with a card. Later a moving sphere is placed in this scene.

```python3
with ui.page('/test') as page:
    card = ui.card()

with card:
    with ui.scene() as scene:
        sphere1 = scene.sphere()
        ui.timer(0.1, lambda: sphere1.move(x=time.time() % 2 - 1))
```

But the movement is not visible, because UI updates are sent to clients of the current page.
And the current page is "/" and not "/test", because the sphere is defined outside of `page`.

Maybe the problem is in `Element.__init__`:

```python3
self.parent_view = view_stack[-1]
self.page = page_stack[-1]
```

Although the `sphere` is placed within the parent view `card`, its page is read from the global `page_stack`.
It could instead use the page from its parent view and the inconsistency is gone.
If this solves the issue, we should also check other accesses to the `page_stack` and if they are definitely correct.

As a workaround one can make the page context explicit: `with page, card:`. But we should improve the behavior nonetheless.

**Comment by falkoschindler** [comment]
Inheriting the page from the parent view seems to work indeed. The sphere is moving. The only stumbling block is the fact that JustPy views can have multiple pages. But within NiceGUI we can assume to have exactly one page per view.

https://github.com/zauberzeug/nicegui/commit/96449f234f513e492797243a61dc3486835cd27c

**Comment by rodja** [comment]
Released in [v0.8.8](https://github.com/zauberzeug/nicegui/releases/tag/v0.8.8)
